sql/src/main/java/io/crate/expression/reference/sys/cluster/ClusterLicenseExpression.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.expression.reference.sys.cluster;
+
+import io.crate.expression.reference.NestedObjectExpression;
+import io.crate.license.DecodedLicense;
+import io.crate.license.DecryptedLicenseData;
+import io.crate.license.LicenseKey;
+import io.crate.license.LicenseService;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.logging.Loggers;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ClusterLicenseExpression extends NestedObjectExpression {
+
+    private static final Logger logger = Loggers.getLogger(ClusterLicenseExpression.class);
+    private final LicenseService licenseService;
+
+    public static final String NAME = "license";
+
+    @Inject
+    public ClusterLicenseExpression(LicenseService licenseService) {
+        this.licenseService = licenseService;
+    }
+
+    @Override
+    public Map<String, Object> value() {
+        HashMap<String, Object> licenseValues = new HashMap<>();
+        LicenseKey licenseKey = licenseService.currentLicense();
+        if (licenseKey != null) {
+            DecodedLicense decodedLicense = LicenseKey.decodeLicense(licenseKey);
+            licenseValues.put("license_type", decodedLicense.type());
```

Comment:
I think we should keep the type and version internal and only expose the actual contents.

I'm also wondering if the licenseService should keep the decrypted license in memory so that we don't have to run this logic on each request

Reply:
I think we should expose the type and version even as just a means for us to debug if something is wrong with license validation.

I'm not sure about decrypting it. I don't think people automate `sys.cluster` queries, especially selecting `*` or `license` but don't have any strong opinion (we'll have to decrypt it and run this logic on cluster change is we decide to hold the information decrypted which makes me a bit nervous)

Reply:
> I think we should expose the type and version even as just a means for us to debug if something is wrong with license validation

I disagree. It's leaking implementation details - that there are 2 different variants of licenses - to the user and I think that's not something we should do.

> I'm not sure about decrypting it. I don't think people automate sys.cluster queries, 

We'll have a sys.check that requires the expiration date information and that is periodically queried by the admin-ui. Decryption (especially asymmetric) is relatively expensive, so I think some kind of caching within the LicenseService would be justified. But we can also do some benchmarks first to see what we're talking about

Reply:
Right. Ok, I'll make the changes.